### PR TITLE
Add support for Observability with minimal manifest

### PR DESCRIPTION
**Pull request by AkshaySainiDell**
# Description
Add support for Observability with minimal manifest for PowerFlex and PowerScale driver.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1449 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Ran PowerScale with Observability E2E, metrics and topology tests
metrics
topology

- [x] Ran PowerFlex with Observability E2E, metrics and topology tests
image

image

### CODE REVIEW

**Review by suryagupta4 — DISMISSED** [review]

File: operatorconfig/moduleconfig/observability/v1.10.0/karavi-metrics-powerflex.yaml [hunk 1]
```diff
@@ -105,7 +105,7 @@ spec:
       serviceAccount: karavi-metrics-powerflex-controller
       containers:
         - name: karavi-metrics-powerflex
-          image: <POWERFLEX_OBS_IMAGE>
+          image: dellemc/csm-metrics-powerflex:v1.7.0
```
Comment (meggm) [inline operatorconfig/moduleconfig/observability/v1.10.0/karavi-metrics-powerflex.yaml:108]: What is the reason for changing the image parameter placeholder to the image name??

Comment (AkshaySainiDell) [reply operatorconfig/moduleconfig/observability/v1.10.0/karavi-metrics-powerflex.yaml:108]: It is handled the same way for all other components.

**Review by meggm — DISMISSED** [review]

File: pkg/modules/observability.go [hunk 2]
```diff
@@ -284,21 +267,19 @@ func ObservabilityTopology(ctx context.Context, isDeleting bool, op utils.Operat
...
 
 	buf, err := readConfigFile(obs, cr, op, TopologyYamlFile)
 	if err != nil {
-		return YamlString, err
+		return nil, err
 	}
-	YamlString = string(buf)
+	YamlString := string(buf)
 
+	fmt.Println(string(buf))
```
Comment (adarsh-dell) [inline pkg/modules/observability.go:282]: Do we need this `println `statement here?

Comment (AkshaySainiDell) [reply pkg/modules/observability.go]: Good observation. Removed it.

**Review by adarsh-dell — APPROVED** [review]
lgtm

**Review by suryagupta4 — APPROVED** [review]

**Review by meggm — APPROVED** [review]

**Pull request merged**